### DISCUSSION TITLE
best way to calculate spyrmsd for ligands with different number of atoms?

### DISCUSSION THREAD

**Discussion by zoeyxyang** [RMeli/spyrmsd#142] [Q&A]
what is the best way to calculate spyrmsd when the ligands don't have the same number of atoms?
For example, if I have a pdb with certain ligand that has some missing atoms in the structure, and I need to calculate the rmsd between this ground truth ligand and the output of a docked ligand  (which is the same ligand but has no missing atoms).
In this case spyrmsd will give an error: coords1.shape == coords2.shape

Is there a way for spyrmsd to only calculate the rmsd on the overlapping atoms between these two ligands?

**Comment by RMeli** [comment]
Hi @zoeyxyang, this is not supported by `spyrmsd`. It sounds to me a bit of a corner case, which I have not encounter before. You will have to manually post-process your docked ligands to look like (also from a molecular graph perspective) reference one.

An alternative would be to change the graph isomorphism into sub-graph isomorphism in the code base, which can match incomplete graphs. However, this would be a bit more involved to implement than a simple post-processing.

**Reply by zoeyxyang** [reply]
Thanks @RMeli 
I was able to fix this issue by adding the missing atoms using Rosetta.
So, now I have two molecules with the same atoms.
However, when I calculate rmsd on these two structures (they are in pdb format, I'm changed to txt so I can upload it here) :
[CDL1_alignLipid.txt](https://github.com/user-attachments/files/18043113/CDL1_alignLipid.txt)
[lipid_fix_noH.txt](https://github.com/user-attachments/files/18043114/lipid_fix_noH.txt)

,even thought their connectivity looks the same to me, spyrmsd gives an error: "pyrmsd.exceptions.NonIsomorphicGraphs: Graphs are not isomorphic.
Make sure graphs have the same connectivity."


Any ideas on why it could be?
Thank you so much!

**Reply by RMeli** [reply]
Hi @zoeyxyang, I had a quick look and the PDB files seem indeed fine. However, I tried to use `obabel` to convert the two files to SDF and to my surprise, one of the files has 100 bonds, the other 99. Therefore, I suspect that Open Babel or RDKit (depending on what you are using) are changing the molecule when loading it. I'll try to find a bit more time in the coming days to look at the issue in more details, but please keep me updated if you find something on your end.

**Reply by zoeyxyang** [reply]
Hello @RMeli , thanks for investigating in this.
Just for your reference, I used RDKit for the inputs. So it seems like it may have the same issue with obabel.
